File: frontend/src/api/video.js
import request from './index'

/**
 * 提交视频生成任务
 * @param {Object} params
 * @param {string} params.poem_text 诗词文本
 * @param {string} params.title 诗词标题
 * @param {string} params.style 视频风格
 */
export function submitVideoTask(params) {
  return request.post('/video/generate', params, { timeout: 60000 })
}

/**
 * 查询视频生成状态
 * @param {string} taskId 任务 ID
 */
export function queryVideoStatus(taskId) {
  return request.get(`/video/status/${taskId}`, { timeout: 30000 })
}

File: backend/app/routers/video.py
"""诗境动画路由"""
from fastapi import APIRouter
from app.models import VideoGenerateRequest
from app.utils import Result
from app.services import video_service

router = APIRouter(prefix="/api/video", tags=["诗境动画"])


@router.post("/generate", summary="提交视频生成任务", response_model=Result)
async def generate_video(req: VideoGenerateRequest):
    """提交诗词文生视频任务，返回 task_id 用于轮询"""
    try:
        result = await video_service.submit_video_task(
            poem_text=req.poem_text,
            title=req.title,
            style=req.style,
        )
        return Result.success(result)
    except Exception as e:
        import logging
        logging.getLogger(__name__).warning("Video generation submit failed: %s", e)
        return Result.error("视频生成任务提交失败，请稍后再试")


@router.get("/status/{task_id}", summary="查询视频生成状态", response_model=Result)
async def video_status(task_id: str):
    """轮询视频生成任务状态"""
    try:
        result = await video_service.query_video_task(task_id)
        return Result.success(result)
    except Exception as e:
        import logging
        logging.getLogger(__name__).warning("Video status query failed: %s", e)
        return Result.error("查询视频状态失败，请稍后再试")
